bioptim/limits/penalty.py

```diff
@@ -578,10 +609,10 @@ def minimize_muscles_control(penalty: PenaltyOption, pn: PenaltyNodes):
                     pn, target, combine_to="muscles_control", axes_idx=Mapping(muscles_idx)
                 )
 
-            # Add the nbTau offset to the muscle index
-            muscles_idx_plus_tau = [idx + pn.nlp.shape["tau"] for idx in muscles_idx]
-            for i, v in enumerate(pn.u):
-                val = v[muscles_idx_plus_tau]
+            V, first_idx, idx = PenaltyFunctionAbstract._get_states_or_controls_with_specified_var(pn, 'muscles')
```

Comment:
This will fail since muscles can be in both controls and states. That said, since the function is called `minimize_muscles_control`, there is no need to dynamically find if we need to minimize the muscles as states or as controls

Reply:
You're right, plus it is useless yet since, MUSCLE_EXCITATIONS_AND_TORQUE_DERIVATIVE_DRIVEN still need to be coded.